core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -357,6 +469,142 @@ public void register(ActionHandler<?, ?> handler) {
         return unmodifiableList(filters);
     }
 
+    private Set<Class<? extends RestHandler>> setupRestHandlers() {
+        ExtensionPoint.ClassSet<RestHandler> handlers = new ExtensionPoint.ClassSet<>("rest_handler", RestHandler.class);
```

Comment:
Yeah. I reused it because it has warnings if you try to bind two copies. I'm a bit conflicted about that because when we move to a "build it in the plugin" style thing it won't work....